iohk-monitoring/test/Cardano/BM/Test/Monitoring.lhs

```diff
@@ -173,3 +195,80 @@ parseEvalExpression t res env =
         Just e  -> evaluate env e @?= res
 
 \end{code}
+
+\subsubsection{Actions tests}
+
+\begin{code}
+monitoringThr :: Trace IO Text -> IO (Async.Async ())
+monitoringThr trace = do
+    trace' <- appendName "monitoring" trace
+    proc <- Async.async $ sendTo trace'
+    return proc
+  where
+    sendTo tr = do
+        (,) <$> (mkLOMeta Warning Public)
+            <*> pure (LogValue "monitMe" (PureI 100))
+        >>= traceNamedObject tr
+
+testSetGlobalMinimalSeverity :: Assertion
+testSetGlobalMinimalSeverity = do
+    let initialGlobalSeverity = Debug
+        targetGlobalSeverity  = Info
+
+    c <- CM.empty
+    CM.setMinSeverity c initialGlobalSeverity
+    CM.setDefaultBackends c [MonitoringBK]
+    CM.setSetupBackends c [MonitoringBK]
+
+    CM.setBackends c "complex.monitoring.monitMe" (Just [MonitoringBK])
+
+    CM.setMonitors c $ HM.fromList
+        [ ( "complex.monitoring"
+          , ( Nothing
+            , Compare "monitMe" (GE, (OpMeasurable 10))
+            , [SetGlobalMinimalSeverity targetGlobalSeverity]
+            )
+          )
+        ]
+
+    (tr' :: Trace IO Text, _) <- setupTrace_ c "complex"
```

Comment:
could `setupTrace` do the job here? it looks like you do not need to access the switchboard from your tests.

Reply:
> it looks like you do not need to access the switchboard from your tests.

No, I _need_ access the `Switchboard`, because real action occurs in the `Switchboard`: https://github.com/input-output-hk/iohk-monitoring-framework/blob/master/iohk-monitoring/src/Cardano/BM/Output/Switchboard.lhs#L214-L219